cmake/tpl/FindKLU.cmake

```diff
@@ -30,6 +30,18 @@
 #   KLU_LIBRARIES   - all of the libraries needed for KLU
 # ---------------------------------------------------------------
 
+if (NOT (KLU_INCLUDE_DIR OR KLU_LIBRARY_DIR OR KLU_LIBRARY))
+  # prefer import target from upstream SuiteSparse if available and
+  # user didn't point to a specific (different) version
+  include ( CMakeFindDependencyMacro )
+  find_dependency(KLU CONFIG)
+
+  if(TARGET SuiteSparse::KLU)
+    add_library(SUNDIALS::KLU ALIAS SuiteSparse::KLU)
+    return()
+  endif()
+endif()
+
 # Set library prefixes for Windows
 if(WIN32)
```

Comment:
Unrelated to this change. But for MinGW, setting the library prefix is not necessary and setting the library suffix to d.lib is incorrect. The static library suffix is `.a` for MinGW.
``` suggestion
if(MSCV)
```

Would it be ok to change that in the same PR?

Reply:
Prepending those values to the list of library prefixes and suffixes might not actually cause an issue if no libraries are found that match those patterns. But slight ABI differences between MSVC and MinGW could lead to unexpected behavior when mixing binaries of both targets.

Reply:
Yeah, that change makes sense to include here.